### ISSUE TITLE
draft comments?

### ISSUE THREAD

**Issue by germasch** [fisiceoir/HPC2019-Project1-KdV#10]
So I'm glad to see you guys are using github issues etc. I see a lot of branches, though, so I didn't really know where to look for a version, if any to comment on. Let me know if there is a draft you'd like to get feedback on.

**Comment by jefflynn1** [comment]
Hi Kai!

The testing branch will be best to check out our work. We are meeting later today though, so that might change!

**Comment by AlexanderKnight** [comment]
Hey @germasch. Jeff and I have been debugging the code, and we are having an issue of the data growing uncontrollably as time progresses. As far as we can tell, the code looks good, but something is going wrong, and we can't seem to find it.

**Comment by germasch** [comment]
So if things are just blowing up, it could be as easy as the timestep being
too large (you want something like \Delta t / \Delta x ~ 1), but of course
it could be some bug in your code, too.

--Kai

**Comment by AlexanderKnight** [comment]
So we have made our timestep to be very small, but that is not really the problem, we think. It looks more like there is some error at the boundary points. I will attach several photos at later and later times.

**Comment by AlexanderKnight** [comment]
![plot0](https://user-images.githubusercontent.com/15788315/55765581-5b6f3e00-5a3e-11e9-9dc0-74a9d5f0a9c1.png)
![plot150](https://user-images.githubusercontent.com/15788315/55765582-5b6f3e00-5a3e-11e9-8c1b-3babb1e04ee4.png)
![plot298](https://user-images.githubusercontent.com/15788315/55765583-5b6f3e00-5a3e-11e9-817f-d12a350e8303.png)

**Comment by AlexanderKnight** [comment]
Those are at times 0, 150, and 300.

**Comment by AlexanderKnight** [comment]
And the timestep is about 1e-6 seconds, with dx=0.02

**Comment by jefflynn1** [comment]
@germasch our main.cpp calls timestep.cpp, which calls rk4.cpp, which calls func.cpp. 

We have a 2D array. Each row of our 2d array is an iteration of timestep. The first row is before timestepping (original data), row 1 is after the first call, and so on. 

We think updating the values from timestep into our 2d array is correct. We think our rk4 file is correct mathematically, and func. 

The boundary problems we are having leads me to believe timestep.cpp is to blame, but I think we took proper measures to make sure we don't get bad values. 

main -> updates row of 2d array by calling stepping
stepping ->  takes input array (2d[i-1]) and updates output array  (2d[i]) by calling rk4 for each element of input array
rk4 -> calls func and calculates k values
func -> given 5 points, outputs updated point

**Comment by germasch** [comment]
I agree that it looks like a boundary problem (however, it's hard to be
sure). Boundary problems usually get worse if you use lower resolution, so
that might help to identify it even more clearly, in particular if it's
some kind of off-by-one problem. Another thing that might help to track
this down is to use a simpler version of the equation you're solving, e.g.,
just use "c du/dx" which is just a simple wave equation.

--Kai

**Comment by AlexanderKnight** [comment]
@germasch, would you be available to meet after class tomorrow? We implemented a couple of your suggestions, but did not end up fixing the problem.

**Comment by germasch** [comment]
Actually, I won't have time tomorrow after class, but I'd have time before,
or in the morning.

--Kai

**Comment by jefflynn1** [comment]
Hi @germasch , I unfortunately have work until class. I just found out I got another job phone interview today so I won't be in class today. @fisiceoir Do you want to keep working on the project today? I will have some time tonight.

**Comment by AlexanderKnight** [comment]
@germasch 

We are both still having major issues. Our code is still going unstable, even if we offset the initial sine wave. We have looked at just about every single calculation and process, and it seems like the problem is always starting at the edges. I recoded the cyclic boundary, and still got the same results. I wanted to check that our numerical equation of state was accurate, since that is the only thing that I cannot be totally sure of.
![32qnow](https://user-images.githubusercontent.com/15788315/56058467-776d2b00-5d2f-11e9-9b80-2c1f5e065a99.png)
Is this equation correct?

**Comment by germasch** [comment]
It looks correct to me. If the problem starts at the boundary, the points
to a boundary issue, still. If you're around, you could try to find me in
my office at/after 4pm.

--Kai
